### PR TITLE
Add support for transOrderBy

### PR DESCRIPTION
**Pull request by aurelien-roy**
These changes add a new method enabling models to be sorted by an indexed translatable field.

Use case : Print a list of items sorted by their name where name is a translatable field. Of course, sorting won't be the same from one language to another.

Example :

```php
Product::transOrderBy('name')->get();
```

If a record hasn't be translated, the value from default locale will be used for sorting.

### CODE REVIEW

File: behaviors/TranslatableModel.php [hunk 1]
```diff
@@ -48,6 +44,47 @@ public function scopeTransWhere($query, $index, $value, $locale = null, $operato
...
+
+    /**
+     * Joins the translatable indexes table to a query.
+     * This cannot be executed more than once.
+     * @param  Builder $query
+     * @param  string $locale
+     * @return Builder
+     */
+    protected function joinTranslateIndexesTable($query, $locale = null) {
+
```
Comment (LukeTowers) [inline behaviors/TranslatableModel.php]: get rid of this extra line

**Comment by LukeTowers** [comment]
@mjauvin is on vacation for the next few weeks so unless someone else reviews it it'll take a few weeks to get to this.

File: tests/unit/behaviors/TranslatableModelTest.php [hunk 2]
```diff
@@ -101,4 +102,52 @@ public function testSetTranslationValue()
...
+
+        $obj->translateContext('fr');
+        $obj->name = 'Australie';
+        $obj->save();
+
+        $this->assertEquals(0, CountryModel::transWhere('name', 'Australie')->count());
+
+        Translator::instance()->setLocale('fr');
+        $this->assertEquals(1, CountryModel::transWhere('name', 'Australi')->count());
+
```
Comment (mjauvin) [inline tests/unit/behaviors/TranslatableModelTest.php]: Is that a typo, should it be "Australie" instead of "Australi" ?

Comment (aurelien-roy) [reply tests/unit/behaviors/TranslatableModelTest.php]: Indeed^^

**Comment by mjauvin** [comment]
@LukeTowers looks good to me!

**Comment by LukeTowers** [comment]
@mjauvin I don't see anything that stands out to me so if you've tested it and it works well feel free to merge when appropriate

**Comment by mjauvin** [comment]
@LukeTowers @bennothommo I've tested this with the following OctoDock instance and it's working fine as far as I can see:

   https://d3f3b84cf746.octodock.com/backend

I also ran the phpunit tests within the instance without issues.

I'm ready to merge this if you have nothing further to add.

**Comment by bennothommo** [comment]
@mjauvin Looks fine to me. It's new functionality, so its introduction shouldn't break anything.

**Comment by aurelien-roy** [comment]
Thank you all for your time reviewing it!

File: behaviors/TranslatableModel.php [hunk 3]
```diff
@@ -33,7 +33,7 @@ public function __construct($model)
 
     /**
      * Applies a translatable index to a basic query. This scope will join the index
-     * table and cannot be executed more than once.
+     * table and can be executed neither more than once, nor with scopeTransOrder.
```
Comment (LukeTowers) [inline behaviors/TranslatableModel.php]: @mjauvin what does this mean?

Comment (mjauvin) [reply behaviors/TranslatableModel.php]: Indeed... @aurelien-roy can you fix this comment so it makes sense?

**Pull request merged**